File: prev codes/tcpserver.py
import socket 

server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
server.bind(("0.0.0.0", 5001))
server.listen(1)

client, addr = server.accept()

while True:
    clientMsg = client.recv(1024).decode('utf-8')
    
    if clientMsg == "quit":
        print("disconnected")
    else:
        print(f"Client: {clientMsg}")

    serverMsg = input("You: ")
    client.sendall(serverMsg.encode('utf-8')) 

server.close()
client.close()

File: prev codes/TCPServer.java
import java.net.*;
import java.io.*;

public class TCPServer {
    public static void main(String[] args) throws IOException {
        ServerSocket server = new ServerSocket(5001);
        Socket client = server.accept();
        BufferedReader in = new BufferedReader(new InputStreamReader(client.getInputStream()));
        PrintWriter out = new PrintWriter(client.getOutputStream(), true);
        BufferedReader userIn = new BufferedReader(new InputStreamReader(System.in));

        System.out.println("Client connected!");
        String clientMsg, serverMsg;
        while (true) {
            clientMsg = in.readLine();
            if (clientMsg.equalsIgnoreCase("quit")) {
                System.out.println("Client disconnected.");
                break;
            }
            System.out.println("Client: " + clientMsg);

            System.out.print("You: ");
            serverMsg = userIn.readLine();
            out.println(serverMsg);
        }

        client.close();
        server.close();
    }
}
